submariner/multiple-cable-drivers.md

```diff
@@ -0,0 +1,415 @@
+# Support fine-grained cluster connectivity and multiple cable drivers
+
+## Summary
+
+Submariner currently assumes full mesh connectivity by default and provisions VPN connections between all participating
```

Comment:
I suggest for sake of simplicity to stick with full mesh for this discussion as other topologies have been already discussed before and pose little benefit but come at great costs.

Reply:
If we are just using full mesh then the rest of this proposal is null and void, no? as we are already operating in the desired mode

Reply:
There is value in allowing different types of tunneling (or encryption) to happen within parts of the clusterset, I don't see how that's null and void.

Reply:
So should this be 2 proposals?

Reply:
Again @mkolesnik might be referring to the gateway topologies which this proposal isn't referring to.

However this proposal does combine 2 concepts which may or may not be disparate, I think depending on one's POV or use case: 1) should two clusters establish a connection and 2) how should they be connected (ie which cable driver and config to use). The absolute original proposal also combined them using the same CRD, as you've done here. But after much discussion, it was decided that the concepts and known associated use cases were different and thus it made sense to defined separate CRDs, hence the [second proposal](https://github.com/submariner-io/enhancements/pull/4). I think it might make sense to continue down that path. It would allow us to implement one w/o the other. I think you're really interested in 2) at this point anyway. The semantics of the cable driver connection config policy seem much simpler than that of the connectivity policy and I think more thought and discussion and perhaps more real-world uses cases are needed.

Reply:
No worries at all. Continuing down a path that the community agreed made sense. 

Can I close this proposal?

Reply:
Sure however you want to do it.